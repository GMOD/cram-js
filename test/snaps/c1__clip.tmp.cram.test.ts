import { testFile } from '../testUtil'
testFile('c1#clip.tmp.cram')
