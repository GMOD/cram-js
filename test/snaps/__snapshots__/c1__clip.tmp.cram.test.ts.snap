// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can dump the whole c1#clip.tmp.cram without error 1`] = `
[
  {
    "fileId": "-",
    "magic": "CRAM",
    "majorVersion": 3,
    "minorVersion": 0,
  },
  {
    "containerHeader": {
      "_endPosition": 44,
      "_size": 18,
      "alignmentSpan": 0,
      "crc32": 3519680774,
      "landmarks": [
        0,
        119,
      ],
      "length": 187,
      "numBases": 0,
      "numBlocks": 2,
      "numLandmarks": 2,
      "numRecords": 0,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 0,
    },
    "data": [
      {
        "_endPosition": 44,
        "_size": 18,
        "alignmentSpan": 0,
        "crc32": 3519680774,
        "landmarks": [
          0,
          119,
        ],
        "length": 187,
        "numBases": 0,
        "numBlocks": 2,
        "numLandmarks": 2,
        "numRecords": 0,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 0,
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "SN",
                "value": "c1",
              },
              {
                "tag": "LN",
                "value": "10",
              },
              {
                "tag": "M5",
                "value": "e224bd8a70e5466c6a033be886045c85",
              },
              {
                "tag": "UR",
                "value": "/Users/rbuels/dev/htslib/test/c1.fa",
              },
            ],
            "tag": "SQ",
          },
        ],
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "SN",
                "value": "c1",
              },
              {
                "tag": "LN",
                "value": "10",
              },
              {
                "tag": "M5",
                "value": "e224bd8a70e5466c6a033be886045c85",
              },
              {
                "tag": "UR",
                "value": "/Users/rbuels/dev/htslib/test/c1.fa",
              },
            ],
            "tag": "SQ",
          },
        ],
      },
    ],
  },
  {
    "compressionScheme": {
      "APdelta": true,
      "dataSeriesEncoding": {
        "AP": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 17,
          },
          "parametersBytes": 1,
        },
        "BB": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 42,
              },
              "parametersBytes": 1,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 37,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 6,
        },
        "BF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "CF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              3,
            ],
          },
          "parametersBytes": 4,
        },
        "FC": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 27,
          },
          "parametersBytes": 1,
        },
        "FN": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 26,
          },
          "parametersBytes": 1,
        },
        "FP": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 28,
          },
          "parametersBytes": 1,
        },
        "HC": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 36,
          },
          "parametersBytes": 1,
        },
        "IN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 13,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "MF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "MQ": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "NP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "NS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "QS": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 12,
          },
          "parametersBytes": 1,
        },
        "RG": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "RI": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "RL": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 25,
          },
          "parametersBytes": 1,
        },
        "RN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 11,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "RS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              4,
            ],
          },
          "parametersBytes": 4,
        },
        "SC": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 14,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "TL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "TS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
      },
      "readNamesIncluded": true,
      "referenceRequired": false,
      "substitutionMatrix": [
        [
          "C",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "T",
        ],
      ],
      "tagEncoding": {},
      "tagIdsDictionary": [
        [],
      ],
    },
    "containerHeader": {
      "_endPosition": 249,
      "_size": 18,
      "alignmentSpan": 10,
      "crc32": 2292045856,
      "landmarks": [
        194,
      ],
      "length": 632,
      "numBases": 62,
      "numBlocks": 13,
      "numLandmarks": 1,
      "numRecords": 7,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 1,
    },
    "data": [
      {
        "_endPosition": 249,
        "_size": 18,
        "alignmentSpan": 10,
        "crc32": 2292045856,
        "landmarks": [
          194,
        ],
        "length": 632,
        "numBases": 62,
        "numBlocks": 13,
        "numLandmarks": 1,
        "numRecords": 7,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 1,
      },
      {
        "features": [
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 10,
            "mappingQuality": 0,
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 10,
            "readName": "s0a",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 449,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 10,
            "mappingQuality": 0,
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "N",
                "data": 4,
                "pos": 4,
                "refPos": 4,
              },
            ],
            "readGroupId": -1,
            "readLength": 6,
            "readName": "s0A",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 450,
          },
          {
            "alignmentStart": 2,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 8,
            "mappingQuality": 0,
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "S",
                "data": "A",
                "pos": 1,
                "refPos": 2,
              },
              {
                "code": "S",
                "data": "T",
                "pos": 10,
                "refPos": 10,
              },
            ],
            "readGroupId": -1,
            "readLength": 10,
            "readName": "s0b",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 451,
          },
          {
            "alignmentStart": 2,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 8,
            "mappingQuality": 0,
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "H",
                "data": 1,
                "pos": 1,
                "refPos": 2,
              },
              {
                "code": "H",
                "data": 1,
                "pos": 9,
                "refPos": 10,
              },
            ],
            "readGroupId": -1,
            "readLength": 8,
            "readName": "s0B",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 452,
          },
          {
            "alignmentStart": 3,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 6,
            "mappingQuality": 0,
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "S",
                "data": "AA",
                "pos": 1,
                "refPos": 3,
              },
              {
                "code": "S",
                "data": "TT",
                "pos": 9,
                "refPos": 9,
              },
            ],
            "readGroupId": -1,
            "readLength": 10,
            "readName": "s0c",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 453,
          },
          {
            "alignmentStart": 3,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 6,
            "mappingQuality": 0,
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "S",
                "data": "AA",
                "pos": 1,
                "refPos": 3,
              },
              {
                "code": "I",
                "data": "NN",
                "pos": 6,
                "refPos": 6,
              },
              {
                "code": "S",
                "data": "TT",
                "pos": 11,
                "refPos": 9,
              },
            ],
            "readGroupId": -1,
            "readLength": 12,
            "readName": "s0c",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 454,
          },
          {
            "alignmentStart": 3,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 6,
            "mappingQuality": 0,
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "H",
                "data": 2,
                "pos": 1,
                "refPos": 3,
              },
              {
                "code": "H",
                "data": 2,
                "pos": 7,
                "refPos": 9,
              },
            ],
            "readGroupId": -1,
            "readLength": 6,
            "readName": "s0C",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 455,
          },
        ],
        "header": {
          "_endPosition": 490,
          "_size": 42,
          "compressedSize": 38,
          "compressionMethod": "raw",
          "content": Uint8Array [
            0,
            1,
            10,
            7,
            0,
            11,
            10,
            11,
            12,
            13,
            14,
            17,
            25,
            26,
            27,
            28,
            36,
            255,
            255,
            255,
            255,
            15,
            226,
            36,
            189,
            138,
            112,
            229,
            70,
            108,
            106,
            3,
            59,
            232,
            134,
            4,
            92,
            133,
          ],
          "contentId": 0,
          "contentPosition": 448,
          "contentType": "MAPPED_SLICE_HEADER",
          "crc32": 2766945771,
          "parsedContent": {
            "_endPosition": 287,
            "_size": 38,
            "contentIds": [
              11,
              12,
              13,
              14,
              17,
              25,
              26,
              27,
              28,
              36,
            ],
            "md5": [
              226,
              36,
              189,
              138,
              112,
              229,
              70,
              108,
              106,
              3,
              59,
              232,
              134,
              4,
              92,
              133,
            ],
            "numBlocks": 11,
            "numContentIds": 10,
            "numRecords": 7,
            "recordCounter": 0,
            "refBaseBlockId": -1,
            "refSeqId": 0,
            "refSeqSpan": 10,
            "refSeqStart": 1,
          },
          "uncompressedSize": 38,
        },
      },
    ],
  },
]
`;
